fendermint/app/src/store.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+use std::borrow::Cow;
+
+use fendermint_storage::{Codec, Decode, Encode, KVError, KVResult, KVStore};
+use fvm_ipld_encoding::{de::DeserializeOwned, serde::Serialize};
+
+#[derive(Clone)]
+pub struct AppStore;
+
+impl KVStore for AppStore {
+    type Repr = Vec<u8>;
+    type Namespace = &'static str;
+}
+
+impl<T> Codec<T> for AppStore where AppStore: Encode<T> + Decode<T> {}
+
+/// CBOR serialization.
+impl<T> Encode<T> for AppStore
+where
+    T: Serialize,
+{
+    fn to_repr(value: &T) -> KVResult<Cow<Self::Repr>> {
+        fvm_ipld_encoding::to_vec(value)
+            .map_err(|e| KVError::Codec(Box::new(e)))
+            .map(Cow::Owned)
+    }
+}
+
+/// CBOR deserialization.
+impl<T> Decode<T> for AppStore
+where
+    T: DeserializeOwned,
+{
+    fn from_repr(repr: &Self::Repr) -> KVResult<T> {
+        fvm_ipld_encoding::from_slice(repr).map_err(|e| KVError::Codec(Box::new(e)))
```

Comment:
@adlrocha I can't remember now if the restriction on String keys in maps only affects HAMT fields or everything you ever want to store.

Reply:
I think it's just the HAMT which serializes as CBOR but deserializes as IPLD. Just confusing that we are using `fvm_ipld_encoding` here.